routes/projects.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express');
+const db = require('../data/helpers/projectModel');
+
+
+function validateBody(arr) {
+    return function(req, res, next) {
+        try {
+            arr.map( (i)=>{
+                console.log(i);
+                if(!(typeof req.body[i]) || req.body[i] === "")
+                    throw "invalid request";
+            });
+            next();
+        }
+        catch(e){
+            res.status(400).json({errorMessage: e});
+        }
+    }
+}
+
+const router = express.Router();
+router.get('/', async (req,res) => {
+    try {
+       let result = await db.get();
+        if(result.length === 0) throw "no projects availible";
+        res.status(200).json(result);
+    }
+    catch(e) {
+        await res.status(500).json({errorMessage: e });
+    }
+});
+
+router.get('/:id', async (req,res) => {
+    try {
+        result = await db.get(req.params.id);
+        if((result || result.length === 0)) throw "project not availible";
+        else res.status(200).json(result);
+    }
+    catch(e) {
```

Comment:
The status code for 404 should go above in your check, inside of .catch we usually use 500s, because 500s are for server errors and that's what .catch() catches, server errors. 400s are used for user errors

Reply:
usually I would use a 500, but in this case, the result is coming from a faulty url (pointing to an id that doesn't exist); thus 404 seems appropriate. In all other error cases it will throw a 500 error. (see line 40).

Reply:
but this is what you are supposed to check in the check above, if correct, right now you need to fix it, but in case that the array is empty it means that there is nothing matching that id, and there you'd use the 404, .catch will only be triggered with server errors, thats why we use 500s